src/main/java/ru/sbtqa/tag/pagefactory/drivers/TagWebDriver.java

```diff
@@ -137,11 +136,66 @@ private static void createDriver() throws UnsupportedBrowserException, Malformed
         webDriver.get(WEBDRIVER_STARTING_URL);
     }
 
+    private static void configureDriver(BrowserManager webDriverManager, String browserType) {
+        final String noDriverPathWarnMessage = "The value of property 'webdriver.drivers.path is not specified."
+                + " Trying to automatically download and setup driver.";
+
+        if (!WEBDRIVER_PATH.isEmpty()) {
+            System.setProperty("webdriver." + browserType + ".driver", new File(WEBDRIVER_PATH).getAbsolutePath());
+        } else {
+            LOG.warn(noDriverPathWarnMessage, WEBDRIVER_BROWSER_NAME);
+            configureWebDriverManagerParams(webDriverManager, browserType);
+            webDriverManager.setup();
+        }
+    }
+
+    private static void configureProxy(DesiredCapabilities capabilities) {
+        if (!WEBDRIVER_PROXY.isEmpty()) {
+            setProxy(new BrowserMobProxyServer());
+            proxy.start(0);
+            Proxy seleniumProxy = ClientUtil.createSeleniumProxy(proxy);
+            capabilities.setCapability(CapabilityType.PROXY, seleniumProxy);
+        }
+    }
+
+    private static void configureWebDriverManagerParams(BrowserManager webDriverManager, String browserType) {
+        if (!WEBDRIVER_BROWSER_VERSION.isEmpty() && !browserType.equalsIgnoreCase(BrowserType.IE)) {
+            LOG.info("You have specified 'webdriver.browser.version' property. Trying to find corresponding driver.");
+
+            String mappedVersion = parseDriverVersionFromMapping(WEBDRIVER_BROWSER_VERSION, browserType.toLowerCase());
+            webDriverManager.version(mappedVersion);
+        } else if (!WEBDRIVER_DESIRABLE_VERSION.isEmpty()) {
+            webDriverManager.version(WEBDRIVER_DESIRABLE_VERSION);
+        }
+        if (!WEBDRIVER_NEXUS_LINK.isEmpty()) {
+            webDriverManager.useNexus(WEBDRIVER_NEXUS_LINK);
+        }
+    }
+
+    private static String parseDriverVersionFromMapping(String browserVersion, String browserType) {
+        ClassLoader classLoader = TagWebDriver.class.getClassLoader();
+        try {
+            Path file = Paths.get(classLoader
```

Comment:
To get access to json and inner nodes try to use datajack and ru.sbtqa.tag.datajack.adaptors.datajack-json-adaptor.

use it like this
TestDataObject mapping = new JsonDataObjectAdaptor("src/test/resources/data", "collectionName");
return mapping.get(browserVersion).getValue();

Reply:
Make the path to file as private static final string